### ISSUE TITLE
Adding fstype=1 to an xfs partition

### ISSUE THREAD

**Issue by NickJH** [cosmos72/fstransform#37]
I have an xfs partition created when the fstype default was 0, but I now need it to be set to 1. What is the easiest way of transforming it? Can I do xfs > xfs or do I have to do something like xfs > ext4 > xfs?

Please also flag this as a question rather than an issue.

**Comment by cosmos72** [comment]
Yes, the conversion xfs > xfs has been tested. Although maybe less extensively than other combinations.

And you can pass arbitrary arguments to mkfs.xfs with
```shell
fstransform --opts-mkfs='YOUR_XFS_OPTIONS'   OTHER_OPTIONS_AND_ARGS
```

**Comment by NickJH** [comment]
Thanks. I could not find any xfs tools either. Do I even need to specify options as ftype is the current default (and has been since Centos 7.4?). Will `fstransform \dev\sda3 xfs on its own work or will it just say that it is already xfs, so I'd need the full `fstransform --opts-mkfs='ftype=1' \dev\sda3 xfs` or have I got the syntax completely wrong?

BTW I am editing my question as it seems that `fstype` is a typo and it should be `ftype`.

**Comment by cosmos72** [comment]
Since you say that `ftype=1` is the default, then `fstransform /dev/sda3 xfs` is enough

**Comment by NickJH** [comment]
Unfortunately I have failed so far but I get very close. Running fstransform from GParted produces an unreadable disk. This is probably not GParted's or fstransform's fault as I've seen the same when trying to load other externally formatted disks in ClearOS7 (a Centos7 clone). I don't think you can mount a Centos8 xfs disk in centos7 for the same reason. This is probably an xfs compatibility issue.

I then moved on to:
- Clone a 2TB disk to a 120GB SSD with Clonedisk. This is fine in expert mode as there are only about 38GB of partitions on the 2TB drive
- I booted this disk to see if it worked and it did
- With GParted I moved the swap partition (sda3) to the end of the disk and extended the main partition, /sda2 to take the rest of the disk.
- I booted this disk to see if it worked and it did
- I then put the disk in another ClearOS7 system but did not mount it. I used fstransform v0.9.3-5.el7 from EPEL to convert sda2 from xfs to xfs.
- This appeared to work as I could mount the partition, read its contents and check the ftype which had changed from 0 to 1.
- I put the drive back in the original server which had the 2TB drive, disconnected the 2TB drive then tried to boot it.
- every time it hung early on in the boot after `Reached target Basic System`. From a working boot log, this is a systemd step and the next step is back to the kernel for ahci, but it never gets past this.

Digging a lot further, it appears that fstransform is not preserving the UUID on the partition. In my case, sda1 is /boot and sda2 is / and dracut-initqueue is failing to find the `/dev/disk/by-uuid/5cfe9b8f-97ff-48ba-8b7c-c284992612d5` on the transformed sda2. sda2 now appears to have a different UUID (b9e94856-e39c-4ab5-bc47-8ce490f1d829).

Is there any way of writing this UUID back to the partition? I am not aware of it being possible to write an explicit UUID for a partition, only to generate a new one.

**Comment by cosmos72** [comment]
Yes, the converted filesystem will have a new, auto-generated UUID.

Preserving the filesystem UUID and label (if any) during the conversion could be a useful improvement - it's also quite filesystem dependent. I will have a look at how difficult it would be.

**Comment by NickJH** [comment]
When trying to find out how to change the UUID I found [this article](https://unix.stackexchange.com/questions/12858/how-to-change-filesystem-uuid-2-same-uuid) and it does seem to be very dependent on the filesystem. It looks quite messy for some systems.

**Comment by cosmos72** [comment]
Yes, it's very filesystem dependent. Setting the UUID at creation is probably (slightly) easier.

And yes, it should be in the README. I'll add it
